### PR TITLE
Add ability to check minimal ansible version

### PR DESCRIPTION
**Pull request by ssbarnea**
This will allow us to remove the old `prerun.check_ansible_presence` implementation and make instantiation quite nice an easy.

Also this approach is compatible with a possible future use of ansible-runner.

Partial-Fix: #6

### CODE REVIEW

File: src/ansible_compat/runtime.py [hunk 1]
```diff
@@ -59,13 +60,21 @@ def __init__(
         :param isolated: Assure that installation of collections or roles
                          does not affect Ansible installation, an unique cache
                          directory being used instead.
+        :param version: Minimal version of Ansible required. If not found,
+                        a RuntimError exception is raised.
```
Comment (webknjaz) [inline src/ansible_compat/runtime.py:64]: typo+role
```suggestion
                        a :py:class:`RuntimeError` exception is raised.
```

File: src/ansible_compat/runtime.py [hunk 2]
```diff
@@ -59,13 +60,21 @@ def __init__(
...
         self.project_dir = project_dir or os.getcwd()
         self.isolated = isolated
         if isolated:
             self.cache_dir = get_cache_dir(self.project_dir)
         self.config = AnsibleConfig()
 
+        if version and packaging.version.Version(version) > self.version:
+            raise RuntimeError(
+                "Found incompatible version of ansible runtime %s, instead of %s or newer."
+                % (self.version, version)
```
Comment (webknjaz) [inline src/ansible_compat/runtime.py:75]: We could just use an f-string here, right? If you insist on using placeholders, please make them named (like `%(version)s` for %-formatting or `{version!s}` for `str.format()`).

File: test/test_runtime.py [hunk 3]
```diff
@@ -26,6 +26,12 @@ def test_runtime_version(runtime: Runtime) -> None:
     assert version == runtime.version
 
 
+def test_runtime_version_outdated() -> None:
+    """Checks that instantiation raises if version is outdated."""
+    with pytest.raises(RuntimeError):
```
Comment (webknjaz) [inline test/test_runtime.py:31]: Always use `match=` with `pytest.raises` or `pytest.warns`

File: src/ansible_compat/runtime.py [hunk 4]
```diff
@@ -50,6 +50,7 @@ def __init__(
         self,
         project_dir: Optional[str] = None,
         isolated: bool = False,
+        version: Optional[str] = None,
```
Comment (webknjaz) [inline src/ansible_compat/runtime.py:53]: The docstring says that it's a minimal version. Just version may mean "install/upgrade this version if missing". But if you improve the naming, it may become clear what it is:
```suggestion
        min_required_version: Optional[str] = None,
```

File: src/ansible_compat/runtime.py [hunk 5]
```diff
@@ -59,13 +60,21 @@ def __init__(
...
+        :param version: Minimal version of Ansible required. If not found,
+                        a RuntimError exception is raised.
         """
         self.project_dir = project_dir or os.getcwd()
         self.isolated = isolated
         if isolated:
             self.cache_dir = get_cache_dir(self.project_dir)
         self.config = AnsibleConfig()
 
+        if version and packaging.version.Version(version) > self.version:
```
Comment (webknjaz) [inline src/ansible_compat/runtime.py:72]: "if version" is semantically incorrect and may result in an unexpected control flow. Like if it's `0`.
The correct way of the comparison with `None` is to have an identity check like `if version is not None`.

File: test/test_runtime.py [hunk 6]
```diff
@@ -26,6 +26,12 @@ def test_runtime_version(runtime: Runtime) -> None:
     assert version == runtime.version
 
 
+def test_runtime_version_outdated() -> None:
```
Comment (webknjaz) [inline test/test_runtime.py:29]: I think this test is a good candidate for `@pytest.mark.parametrize` — this way you'll document which types of values you had in mind when adding this feature.

Comment (ssbarnea) [reply test/test_runtime.py:29]: I go for parametrize when there is a clear benefit in avoiding duplication, but for cases like this I prefer not using it. The reason behind is that parameterized tests are harder to read and to run as you need to add extra quotes on shell due to the use of `[...]` in their ids. Anyway, we can easily rewrite them when needed.

Comment (webknjaz) [reply test/test_runtime.py:29]: No need to merge this with other tests. Just add something other than 99, like corner cases.
